src/dynarec/dynarec_arm_66f0.c

```diff
@@ -30,14 +30,69 @@ uintptr_t dynarec66F0(dynarec_arm_t* dyn, uintptr_t addr, uintptr_t ip, int nins
     uint8_t ed, gd, u8;
     int fixedaddress;
     int32_t i32, j32;
+    int16_t i16;
     MAYUSE(i32);
     MAYUSE(j32);
     MAYUSE(gb1);
     MAYUSE(gb2);
     MAYUSE(wb1);
     MAYUSE(wb2);
     switch(opcode) {
-        
+        case 0x81:
+        case 0x83:
+            nextop = F8;
+            DMB_ISH();
+            switch((nextop>>3)&7) {
+                case 0: //ADD
+                    if(opcode==0x81) {
+                        INST_NAME("LOCK ADD Ew, Iw");
+                    } else {
+                        INST_NAME("LOCK ADD Ew, Ib");
+                    }
+                    SETFLAGS(X_ALL, SF_SET_PENDING);
+                    DMB_ISH();
+                    if((nextop&0xC0)==0xC0) {
+                        if(opcode==0x81) i16 = F16S; else i16 = F8S;
+                        ed = xEAX+(nextop&7);
+                        emit_add16c(dyn, ninst, ed, i16, x3, x14);
+                    } else {
+                        addr = geted(dyn, addr, ninst, nextop, &wback, x2, &fixedaddress, 0, 0, 1, LOCK_LOCK);
+                        if(opcode==0x81) i16 = F16S; else i16 = F8S;
+                        if(!fixedaddress) {
+                            TSTS_IMM8(wback, 0x3);
```

Comment:
This should be `TSTS_IMM8(wback, 0x1);` as this is 2-bytes test alignment, not a 4-bytes one